### PR TITLE
Home Page UI Revamp

### PR DESCRIPTION
**Pull request by SmitGala**

### CODE REVIEW

File: app/components/file-compare/file-overview/header/index.hbs [hunk 1]
```diff
@@ -18,20 +18,31 @@
...
-        @color='textSecondary'
+        @variant='body1'
         data-test-fileCompare-fileOverview-packageName
       >
         {{this.packageName}}
       </AkTypography>
     </AkStack>
   </AkStack>
 
+  {{#if @showMenunButton}}
```
Comment (future-pirate-king) [inline app/components/file-compare/file-overview/header/index.hbs:35]: fix typo

File: app/components/file-compare/file-overview/file-details/index.hbs [hunk 2]
```diff
@@ -24,25 +24,29 @@
...
+    <AkStack data-test-fileCompare-fileOverview-platformIcon>
+      {{#if @file.project.isIos}}
+        <AkSvg::IosIcon />
+      {{/if}}
 
-    {{! <AkSvg::AmPlaystoreLogo /> }}
+      {{#if @file.project.isAndroid}}
+        <AkSvg::AndroidIcon />
+      {{/if}}
+    </AkStack>
```
Comment (future-pirate-king) [inline app/components/file-compare/file-overview/file-details/index.hbs:49]: use existing icons

File: app/components/ak-svg/android-icon.hbs
Comment (future-pirate-king) [inline app/components/ak-svg/android-icon.hbs:1]: remove this svg icon

File: app/components/ak-svg/ios-icon.hbs
Comment (future-pirate-king) [inline app/components/ak-svg/ios-icon.hbs:1]: remove this svg icon

**Review by future-pirate-king — CHANGES_REQUESTED** [review]

File: app/components/ak-text-field/index.hbs [hunk 3]
```diff
@@ -15,13 +18,14 @@
...
     local-class='ak-text-field-root {{if @label "ak-form-label-spacing"}}'
   >
     {{#if (has-block 'leftAdornment')}}
       <div
         data-test-ak-text-field-left-adornment
         local-class='ak-text-field-left-adornment'
+        class='ak-text-field-left-adornment'
       >
         {{yield to='leftAdornment'}}
       </div>
```
Comment (future-pirate-king) [inline app/components/ak-text-field/index.hbs:31]: you can expose class arguments and set it for use

**Review by future-pirate-king — CHANGES_REQUESTED** [review]

File: app/components/project-list/index.ts [hunk 4]
```diff
@@ -4,23 +4,42 @@ import Component from '@glimmer/component';
...
+  @service declare organization: OrganizationService;
+  @service declare store: Store;
+  @service('project') declare projectService: ProjectService;
+
+  @tracked tDateUpdated: string;
+  @tracked tDateCreated: string;
+  @tracked tProjectName: string;
+  @tracked tPackageName: string;
+  @tracked tMostRecent: string;
+  @tracked tLeastRecent: string;
```
Comment (future-pirate-king) [inline app/components/project-list/index.ts:42]: this is not required to be `tracked`

File: app/components/project-list/index.ts [hunk 5]
```diff
@@ -123,11 +164,27 @@ export default class ProjectListComponent extends Component {
...
+        value: -1,
+      },
+      {
+        key: 'Android',
+        value: 0,
+      },
+      {
+        key: 'iOS',
+        value: 1,
+      },
```
Comment (future-pirate-king) [inline app/components/project-list/index.ts:179]: can we use value from `ENUMS`

File: app/components/project-list/index.ts [hunk 6]
```diff
@@ -123,11 +164,27 @@ export default class ProjectListComponent extends Component {
...
+    ];
   }
 
   @action
-  handlePrevNextAction({ limit, offset }) {
+  handlePrevNextAction({ limit, offset }: { limit: number; offset: number }) {
+    console.log(typeof limit);
+    console.log(typeof offset);
```
Comment (future-pirate-king) [inline app/components/project-list/index.ts:186]: remove console.log

File: app/components/project-list/index.ts [hunk 7]
```diff
@@ -142,7 +199,9 @@ export default class ProjectListComponent extends Component {
   }
 
   @action
-  handleItemPerPageChange({ limit }) {
+  handleItemPerPageChange({ limit }: { limit: number }) {
+    console.log(typeof limit);
```
Comment (future-pirate-king) [inline app/components/project-list/index.ts:203]: remove console.log

File: app/components/project-list/index.ts [hunk 8]
```diff
@@ -156,8 +215,11 @@ export default class ProjectListComponent extends Component {
     );
   }
 
-  @action sortProjects(event) {
-    this.sortKey = event?.target?.value;
+  @action sortProjects(event: SortingKeyObject) {
+    console.log(typeof event);
```
Comment (future-pirate-king) [inline app/components/project-list/index.ts:219]: remove console.log

File: app/components/project-list/index.ts [hunk 9]
```diff
@@ -156,8 +215,11 @@ export default class ProjectListComponent extends Component {
     );
   }
 
-  @action sortProjects(event) {
-    this.sortKey = event?.target?.value;
+  @action sortProjects(event: SortingKeyObject) {
```
Comment (future-pirate-king) [inline app/components/project-list/index.ts:218]: rename `event` to `selected`

File: app/components/project-list/index.ts
Comment (future-pirate-king) [inline app/components/project-list/index.ts:1]: register component to component registry

File: app/components/project-overview/index.ts [hunk 10]
```diff
@@ -0,0 +1,27 @@
+// Import the necessary modules and types
```
Comment (future-pirate-king) [inline app/components/project-overview/index.ts:1]: remove comment

File: app/components/project-overview/index.scss [hunk 11]
```diff
@@ -0,0 +1,6 @@
+.no-bg {
+  color: unset;
+  &:hover {
```
Comment (future-pirate-king) [inline app/components/project-overview/index.scss:3]: add new line above

File: app/models/project.ts [hunk 12]
```diff
@@ -111,6 +111,14 @@ export default class ProjectModel extends ModelBaseMixin {
     }
   }
 
+  get isAndroid() {
+    return this.platform === ENUMS.PLATFORM.ANDROID;
+  }
+
+  get isIos() {
+    return this.platform === ENUMS.PLATFORM.IOS;
+  }
```
Comment (future-pirate-king) [inline app/models/project.ts:120]: why is this required ?

**Review by future-pirate-king — CHANGES_REQUESTED** [review]

File: app/components/ak-select/before-option/index.ts [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+    const e = event as InputEvent;
+    if (this.args.onInput(e) === false) {
+      return false;
+    }
+  }
+}
+
+declare module '@glint/environment-ember-loose/registry' {
+  export default interface Registry {
+    AkSelectBeforeOption: typeof AkSelectBeforeOptionComponent;
```
Comment (future-pirate-king) [inline app/components/ak-select/before-option/index.ts:63]: this should be 'AkSelect::BeforeOption'

File: app/components/ak-text-field/index.hbs [hunk 14]
```diff
@@ -31,6 +32,7 @@
       data-test-text-input
       id={{this.id}}
       local-class='ak-text-input'
+      class={{@textInputRootClass}}
```
Comment (future-pirate-king) [inline app/components/ak-text-field/index.hbs:35]: this is not required as this element is exposed

File: app/components/ak-text-field/index.ts [hunk 15]
```diff
@@ -11,6 +11,7 @@ export interface AkTextFieldSignature {
   Args: {
     formControlClass?: string;
     textFieldRootClass?: string;
+    textInputRootClass?: string;
```
Comment (future-pirate-king) [inline app/components/ak-text-field/index.ts:14]: we can remove this

File: app/components/file-overview/file-details/index.hbs [hunk 16]
```diff
@@ -0,0 +1,132 @@
...
+      >
+        <AkTypography
+          @variant='h6'
+          @noWrap={{true}}
+          data-test-fileOverview-versionCode
+        >
+          {{@file.versionCode}}
+        </AkTypography>
+      </AkTooltip>
+
```
Comment (future-pirate-king) [inline app/components/file-overview/file-details/index.hbs:129]: remove this new line

File: app/components/file-overview/file-details/index.hbs [hunk 17]
```diff
@@ -0,0 +1,132 @@
...
+        @title='{{t "version"}} - {{@file.version}}'
+        @arrow={{true}}
+        {{style maxWidth='50px'}}
+      >
+        <AkTypography @variant='h6' @noWrap={{true}}>
+          {{@file.version}}
+        </AkTypography>
+      </AkTooltip>
+    {{else}}
+      <AkSkeleton @width='100px' @height='20px' @variant='rectangular' />
```
Comment (future-pirate-king) [inline app/components/file-overview/file-details/index.hbs:101]: why skeleton when version doesn't exist ?

**Review by future-pirate-king — CHANGES_REQUESTED** [review]

File: app/components/ak-text-field/index.ts (hunk 15 above)
Comment (SmitGala) [reply app/components/ak-text-field/index.ts:14]: My styles were not getting applied if was not adding this

File: app/components/ak-text-field/index.hbs (hunk 14 above)
Comment (SmitGala) [reply app/components/ak-text-field/index.hbs:35]: My styles were not getting applied if was not adding this

File: app/components/ak-select/before-option/index.ts [hunk 18]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+export default class AkSelectBeforeOptionComponent extends Component<AkSelectBeforeOptionArgs> {
+  get optionTitle() {
+    return this.args.extra?.['optionTitle'];
+  }
+
+  get inputClass() {
+    return styles['team-name-text-input'];
+  }
```
Comment (future-pirate-king) [inline app/components/ak-select/before-option/index.ts:26]: not required anymore

File: app/components/ak-text-field/index.hbs [hunk 19]
```diff
@@ -15,13 +15,14 @@
   {{/if}}
 
   <div
-    class={{@textFieldRootClass}}
+    class='{{@textFieldRootClass}}'
```
Comment (future-pirate-king) [inline app/components/ak-text-field/index.hbs:18]: discard this change

File: app/components/file-details/severity-level/index.hbs [hunk 20]
```diff
@@ -15,6 +15,7 @@
     <FileChart
       {{style maxWidth='350px'}}
       @file={{@file}}
+      @profileId={{@profileId}}
```
Comment (future-pirate-king) [inline app/components/file-details/severity-level/index.hbs:18]: this is no longer required

File: app/components/file-details/severity-level/index.ts [hunk 21]
```diff
@@ -7,7 +7,11 @@ export interface FileDetailsSeverityLevelSignature {
   };
 }
 
-export default class FileDetailsSeverityLevelComponent extends Component<FileDetailsSeverityLevelSignature> {}
+export default class FileDetailsSeverityLevelComponent extends Component<FileDetailsSeverityLevelSignature> {
+  get profileId() {
```
Comment (future-pirate-king) [inline app/components/file-details/severity-level/index.ts:11]: this is no longer required

File: app/components/project-list/filter-selected-item/index.ts [hunk 22]
```diff
@@ -0,0 +1,29 @@
+import Component from '@glimmer/component';
+
+interface ProjectListFilterSelectedItemArgs {
+  extra?: Record<string, unknown>;
+}
+
+export default class ProjectListFilterSelectedItemComponent extends Component<ProjectListFilterSelectedItemArgs> {
+  get beforeOptionLabel() {
```
Comment (future-pirate-king) [inline app/components/project-list/filter-selected-item/index.ts:8]: rename this to optionTitle

File: app/components/project-list/header/index.hbs [hunk 23]
```diff
@@ -0,0 +1,162 @@
...
+      </AkStack>
+
+      <AkSelect
+        @onChange={{this.onSortProjectsChange}}
+        @options={{this.sortingKeyObjects}}
+        @selected={{this.selectedSortKey}}
+        @verticalPosition='below'
+        @triggerClass={{this.triggerClass}}
+        @dropdownClass={{this.dropDownClass}}
+        @optionTitle='Sort By'
```
Comment (future-pirate-king) [inline app/components/project-list/header/index.hbs:136]: use translations

File: app/components/project-list/header/index.hbs [hunk 24]
```diff
@@ -0,0 +1,162 @@
...
+          @searchPlaceholder={{t 'searchTeamByName'}}
+          @search={{this.searchTeams}}
+          @searchEnabled={{true}}
+          @options={{this.teams}}
+          @selected={{this.selectedTeam}}
+          @loadingMessage={{t 'loadingTeam'}}
+          @onChange={{this.onSelectTeamChange}}
+          @onOpen={{this.onOpenTFilter}}
+          @tagName='div'
+          @optionTitle='Team'
```
Comment (future-pirate-king) [inline app/components/project-list/header/index.hbs:103]: use translations

File: app/components/project-list/header/index.hbs [hunk 25]
```diff
@@ -0,0 +1,162 @@
...
+    >
+      <AkStack>
+        <AkSelect
+          @onChange={{this.filterPlatformChange}}
+          @options={{this.platformObjects}}
+          @selected={{this.selectedPlatform}}
+          @verticalPosition='auto'
+          @triggerClass={{this.triggerClass}}
+          @dropdownClass={{this.dropDownClass}}
+          @optionTitle='Platform'
```
Comment (future-pirate-king) [inline app/components/project-list/header/index.hbs:65]: use translations

File: app/components/project-list/header/index.ts [hunk 26]
```diff
@@ -0,0 +1,214 @@
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { task } from 'ember-concurrency';
+import IntlService from 'ember-intl/services/intl';
+import OrganizationService from 'irene/services/organization';
```
Comment (future-pirate-king) [inline app/components/project-list/header/index.ts:7]: add new line before

File: app/components/project-list/header/index.ts [hunk 27]
```diff
@@ -0,0 +1,214 @@
...
+    this.selectedTeamName = team.name;
+
+    this.args.onSelectTeam(team);
+  }
+
+  @action onOpenTFilter() {
+    const query = {
+      limit: 10,
+    };
+    this.queryTeams.perform(query);
```
Comment (future-pirate-king) [inline app/components/project-list/header/index.ts:174]: add new line before

File: app/components/project-list/header/index.ts [hunk 28]
```diff
@@ -0,0 +1,214 @@
...
+  }
+
+  @action searchTeams(teamName: string) {
+    if (teamName && teamName.length) {
+      const query = {
+        q: teamName,
+      };
+      this.queryTeams.perform(query);
```
Comment (future-pirate-king) [inline app/components/project-list/header/index.ts:182]: add new line before

File: app/components/project-list/index.hbs [hunk 29]
```diff
@@ -125,33 +56,57 @@
...
+
+      <AkTypography
+        @variant='h5'
+        @fontWeight='regular'
+        @align='center'
+        data-test-no-project-text
+      >
+        {{t 'noProjectExists'}}
+      </AkTypography>
+
```
Comment (future-pirate-king) [inline app/components/project-list/index.hbs:109]: remove this new line

File: app/components/project-list/loader/index.scss [hunk 30]
```diff
@@ -0,0 +1,15 @@
+.loading {
+  border: 1px solid var(--appmonitoring-border-color);
+  border-radius: var(--appmonitoring-border-radius);
```
Comment (future-pirate-king) [inline app/components/project-list/loader/index.scss:3]: change variable names according to component

File: app/styles/_component-variables.scss [hunk 31]
```diff
@@ -93,6 +93,9 @@ body {
   --ak-select-trigger-disabled-background: var(--disabled-background-textfield);
   --ak-select-trigger-arrow-color: var(--text-primary);
 
+  // variables for ak-select-before-option
```
Comment (future-pirate-king) [inline app/styles/_component-variables.scss:96]: rename this to `ak-select/before-option`

File: tests/integration/components/file-overview-test.js [hunk 32]
```diff
@@ -22,24 +22,26 @@ module(
         this.server.create('analysis', { vulnerability: v.id }).toJSON()
       );
 
+      // Profile Model
+      const profile = this.server.create('profile');
+      const normalizedProfile = this.store.normalize(
```
Comment (future-pirate-king) [inline tests/integration/components/file-overview-test.js:27]: add new line above

**Review by future-pirate-king — CHANGES_REQUESTED** [review]

File: app/components/file-overview/file-details/index.ts [hunk 33]
```diff
@@ -4,23 +4,23 @@ import Component from '@glimmer/component';
...
+interface FileOverviewFileDetailsSignature {
   Element: HTMLElement;
   Args: {
     file: FileModel | null;
   };
 }
 
-export default class FileCompareFileOverviewFileDetailsComponent extends Component<FileCompareFileOverviewFileDetailsSignature> {
+export default class FileOverviewFileDetailsComponent extends Component<FileOverviewFileDetailsSignature> {
   @service declare intl: IntlService;
```
Comment (future-pirate-king) [inline app/components/file-overview/file-details/index.ts:15]: remove this it is not used

File: app/components/project-list/loader/index.scss [hunk 34]
```diff
@@ -0,0 +1,15 @@
+.loading {
+  border: 1px solid var(--project-list-border-color);
+  border-radius: var(--project-list-border-radius);
```
Comment (future-pirate-king) [inline app/components/project-list/loader/index.scss:3]: this should be `--project-list-loader`

**Review by future-pirate-king — CHANGES_REQUESTED** [review]

**Review by future-pirate-king — CHANGES_REQUESTED** [review]

File: app/components/project-list/header/index.hbs [hunk 35]
```diff
@@ -0,0 +1,186 @@
...
+            <AkStack @direction='row' @alignItems='center'>
+              <AkIcon
+                @iconName='close'
+                @size='small'
+                local-class='clear-filter-icon'
+              />
+              <AkTypography local-class='clear-filter-label'>
+                {{t 'clearFilter'}}
+              </AkTypography>
+            </AkStack>
```
Comment (future-pirate-king) [inline app/components/project-list/header/index.hbs:147]: use in built named blocks for icon and text

**Review by future-pirate-king — CHANGES_REQUESTED** [review]

File: app/components/project-list/header/index.hbs [hunk 36]
```diff
@@ -0,0 +1,185 @@
...
+            <AkIcon @iconName='search' @color='textSecondary' />
+          </:leftAdornment>
+
+          <:rightAdornment>
+            {{#if @query}}
+              <AkIcon
+                @iconName='close'
+                @color='textSecondary'
+                {{on 'click' this.clearSearchInput}}
+              />
```
Comment (future-pirate-king) [inline app/components/project-list/header/index.hbs:40]: this should be an icon button with variant small

**Review by future-pirate-king — CHANGES_REQUESTED** [review]

**Review by future-pirate-king — APPROVED** [review]
